### PR TITLE
fix(synced-state): auto-reconnect on WebSocket drop

### PR DESCRIPTION
**Pull request by peterp**
## Problem

When a `useSyncedState` WebSocket connection drops (bad wifi, server restart, etc.), the cached RPC client is permanently broken. Subscription callbacks stop firing and the UI shows stale data. The only recovery is a full page reload. This mostly hits read-only consumers (dashboards, vote pages) since they have no outbound writes to surface the error.

Reported in #1131.

## Solution

Hook into capnweb's `onRpcBroken` callback to detect dead connections. When one dies:

1. Evict the broken client from the cache
2. Wait with exponential backoff (1s → 30s cap)
3. Create a fresh WebSocket connection
4. Re-subscribe all active subscriptions that were on the dead client
5. Fetch latest state for each key so consumers catch up on missed updates

No API changes — existing `useSyncedState` consumers get reconnection for free.

### Status callback

Consumers can optionally react to connection changes via `onStatusChange`:

```ts
const useSyncedState = createSyncedStateHook({
  onStatusChange(status) {
    // status: "connected" | "disconnected" | "reconnecting"
  },
});
```

## Test plan

- [x] 17 tests covering reconnection, backoff, re-subscription, status callbacks, and edge cases
- [x] All 498 tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by kltdwrds** [comment]
fyi @benreichman, in case you ever experienced this -- should make driver tracking / auction timing more robust

File: sdk/src/use-synced-state/client-core.ts [hunk 1]
```diff
@@ -43,9 +91,53 @@ if (typeof window !== "undefined") {
...
+function reconnect(endpoint: string, deadClient: SyncedStateClient) {
+  // Don't schedule multiple reconnects for the same endpoint
+  const state = backoffState.get(endpoint) ?? { attempt: 0, timer: null };
+  if (state.timer !== null) {
+    return;
+  }
+
+  notifyStatusChange(endpoint, "disconnected");
+
+  const delayMs = getBackoffMs(state.attempt);
```
Comment (valscion) [inline sdk/src/use-synced-state/client-core.ts:125]: Won't this cause a thundering herd problem if all clients drop and reconnect at the exact same time? I don't know what's the best way to solve it but at least some jitter in addition to exponential backoff could help so the reconnects could be spread out within a scope of e.g. 500ms

**Review by valscion — APPROVED** [review]
Design-wise this approach looks good to me! I don't know how big of a problem the thundering herd would be when DO handles the websocket connections but I do assume that some jitter wouldn't hurt there. Otherwise server restart would cause all clients to reconnect at the exact same time...

File: sdk/src/use-synced-state/client-core.ts (hunk 1 above)
Comment (peterp) [reply sdk/src/use-synced-state/client-core.ts:125]: Each Durable Object is single-threaded and clients are sharded by room, so the thundering herd here is just the clients in one room — not all clients globally. Shouldn't be an issue for realistic use cases.

Still, jitter is easy and good practice, so added it in d4f8e21 — ±25% random jitter on the backoff delay.

Comment (valscion) [reply sdk/src/use-synced-state/client-core.ts:125]: Oh yeah, true. But as you said, the jitter is a nice practice regardless 👍

**Comment by valscion** [comment]
Nice!
